physics/nuclear_networks/interface/include/sphnnet/nuclear_data.hpp

```diff
@@ -162,74 +147,48 @@ struct NuclearDataType : public cstone::FieldStates<NuclearDataType<RealType_, K
         outputFieldIndices = cstone::fieldStringsToInt(outFields, fieldNames);
     }
 
-    /*! @brief resize the number of particles (nuclear)
+    /*! @brief resize the number of particles of detached fields
      *
      * @param size  number of particles to be held by the class
      */
     void resize(size_t size)
     {
-        double growthRate = 1;
-        auto   data_      = data();
-
         devData.resize(size);
 
-        for (size_t i = 0; i < data_.size(); ++i)
-        {
-            if ((!is_hydro(i)) && this->isAllocated(i))
-            {
-                // actually resize
-                std::visit(
-                    [&](auto& arg)
-                    {
-                        size_t previous_size = arg->size();
-
-                        // reallocate
-                        reallocate(*arg, size, growthRate);
-
-                        // fill rho or rho_m1
-                        if ((void*)arg == (void*)(&rho) || (void*)arg == (void*)(&rho_m1))
```

Comment:
There is a `fill` function that you can use in the propagator after resize to initialize certain fields to a specific value.